gammapy/modeling/fit.py

```diff
@@ -895,3 +897,69 @@ def _repr_html_(self):
             return self.to_html()
         except AttributeError:
             return f"<pre>{html.escape(str(self))}</pre>"
+
+
+class FitResults(collections.abc.Sequence):
+    def __init__(self, results):
+        if np.array([not isinstance(result, FitResult) for result in results]).any():
+            raise TypeError(f"Elements in {results!r} are not FitResult objects")
+        self.results = results
+
+    def __add__(self, other):
+        if isinstance(other, FitResult):
+            return FitResult([*self, other])
+        else:
+            raise TypeError(f"Invalid type: {other!r}")
+
+    def __getitem__(self, key):
+        if isinstance(key, np.ndarray) and key.dtype == bool:
+            return self.__class__(list(np.array(self.results)[key]))
+        else:
+            return self.__class__(self.results[key])
+
+    def __len__(self):
+        return len(self.results)
+
+    def success(self):
+        return [f.success for f in self.results]
+
+    def models(self):
+        return Models([model for f in self.results for model in f.models])
+
+    def covariance_results(self):
+        return [f.covariance_result for f in self.results]
+
+    def optimize_results(self):
+        return [f.optimize_result for f in self.results]
+
+    def select_by_interval(self, axis, coord_min, coord_max):
+        if axis and axis.nbin != len(self.results):
```

Comment:
simply passing the `axis` name can also work

Reply:
Wouldn't you need the axis itself to slice?

Reply:
Its not guaranteed that the axis correctly corresponds this fit result. 
I dont know if its safe to allow a slicing like this

Reply:
Ok, so you mean adding the axis name as a parameter of the object? (self.axis_name) And check that the axis name corresponds in case of slicing?